modules/gdscript/gdscript_parser.cpp

```diff
@@ -4659,6 +4676,14 @@ void GDScriptParser::_parse_class(ClassNode *p_class) {
 						break;
 					}
 				}
+				// Try to find variable name shadowing checking from the higher scope class
+				for (ClassNode *class_iterator = current_class->owner; class_iterator; class_iterator = class_iterator->owner) {
```

Comment:
There's no need to check this since inner classes are a different context, so they never shadow the outer class variables (outer class variables aren't accessible from the inner class).